### PR TITLE
S2 t5 calling all the code

### PR DESCRIPTION
**Pull request by alexjames4**
Calling all the code together in line with our class diagram

Check date formatting is linked appropriately.

Code now appears to work. Throws the correct error messages and displays the specific project with required details on the page in line with story 2: 
As a: project manager
I want: to be able to retrieve all information about a specific project
So That: I can see what the project is and who is assigned to it

### CODE REVIEW

File: project.php [hunk 1]
```diff
@@ -5,14 +5,14 @@
 
 try {
     $db = \ProjectManager\Services\DbConnector::connect();
-    $data = \ProjectManager\Hydrators\ProjectsHydrator::getProject($db, $_GET['id']);
-    if ($_GET['id']) {
+    $projectId = $_GET['id'];
+    if (isset($projectId) && is_numeric($projectId)) {
```
Comment (Joshsilas) [inline project.php]: What happens if it is 0 or a negative number

File: src/Hydrators/ProjectsHydrator.php [hunk 2]
```diff
@@ -18,14 +15,11 @@ public static function getProjects(\PDO $db)
         $query->setFetchMode(\PDO::FETCH_CLASS, ProjectEntity::class);
         return $query->fetchAll();
     }
-<<<<<<< HEAD
+
```
Comment (Joshsilas) [inline src/Hydrators/ProjectsHydrator.php]: extra line

File: tests/Services/ConvertToJsonServiceTest.php [hunk 3]
```diff
@@ -14,13 +14,15 @@ public function testConvert_success()
...
+    public function testUnexpectedErrorResponse_success()
     {
-        $result = \ProjectManager\Services\ConvertToJsonService::convert([NAN], 0);
+        $result = \ProjectManager\Services\ConvertToJsonService::UnexpectedErrorResponse();
         $expected = '{"message":"Unexpected error","data":[]}';
         $this->assertSame($expected, $result);
+        $this->assertSame(500, http_response_code());
     }
 
     public function testConvert_failure()
```
Comment (Joshsilas) [inline tests/Services/ConvertToJsonServiceTest.php:28]: General comment - will need unit test for under 0 handle error

File: tests/Services/ConvertToJsonServiceTest.php [hunk 4]
```diff
@@ -14,13 +14,15 @@ public function testConvert_success()
...
 
-    public function testConvert_success_unexpected_error()
+    public function testUnexpectedErrorResponse_success()
     {
-        $result = \ProjectManager\Services\ConvertToJsonService::convert([NAN], 0);
+        $result = \ProjectManager\Services\ConvertToJsonService::UnexpectedErrorResponse();
         $expected = '{"message":"Unexpected error","data":[]}';
         $this->assertSame($expected, $result);
+        $this->assertSame(500, http_response_code());
     }
```
Comment (Joshsilas) [inline tests/Services/ConvertToJsonServiceTest.php:27]: Extra Notes - what if it doesn't exist

File: tests/Services/ConvertToJsonServiceTest.php (hunk 3 above)
Comment (CosminDarius1) [reply tests/Services/ConvertToJsonServiceTest.php:28]: logic method has changed. no longer needed

**Review by Joshsilas — APPROVED** [review]

**Pull request merged**